lib/Sema/TypeChecker.h

```diff
@@ -209,20 +209,20 @@ enum ContextualTypePurpose {
   CTP_EnumCaseRawValue, ///< Raw value specified for "case X = 42" in enum.
   CTP_DefaultParameter, ///< Default value in parameter 'foo(a : Int = 42)'.
 
-  CTP_CalleeResult,     ///< Constraint is placed on the result of a callee.
-  CTP_CallArgument,     ///< Call to function or operator requires type.
-  CTP_ClosureResult,    ///< Closure result expects a specific type.
-  CTP_ArrayElement,     ///< ArrayExpr wants elements to have a specific type.
-  CTP_DictionaryKey,    ///< DictionaryExpr keys should have a specific type.
-  CTP_DictionaryValue,  ///< DictionaryExpr values should have a specific type.
-  CTP_CoerceOperand,    ///< CoerceExpr operand coerced to specific type.
-  CTP_AssignSource,     ///< AssignExpr source operand coerced to result type.
-
-  CTP_CannotFail,       ///< Conversion can never fail. abort() if it does.
+  CTP_CalleeResult,    ///< Constraint is placed on the result of a callee.
```

Comment:
Please don't commit formatting changes like this, it makes it harder to review. Actual change here is addition to `CTP_SubscriptAssignSource`...